include/dump_offload.hpp

```diff
@@ -0,0 +1,518 @@
+            this->doConnect();
+            },
+            "xyz.openbmc_project.Dump.Manager",
+            "/xyz/openbmc_project/dump/" + dumpType + "/entry/" + entryID,
+            "org.freedesktop.DBus.Properties", "Get",
+            "xyz.openbmc_project.Dump.Entry", "Size");
+    }
+
+    /**
+     * @brief  Reads data from unix domain socket and writes on
+     *         http stream connection socket.
+     *
+     * @return void
+     */
+
+    void doReadStream()
+    {
+        std::size_t bytes = outputBuffer.capacity() - outputBuffer.size();
+
+        this->unixSocket.async_read_some(
+            outputBuffer.prepare(bytes),
+            [this, self(shared_from_this())](
+                const boost::system::error_code& ec, std::size_t bytesRead) {
+            if (ec)
+            {
+                if (ec != boost::asio::error::eof)
+                {
+                    BMCWEB_LOG_ERROR << "Couldn't read from local peer: " << ec;
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::internal_server_error);
+                    this->connection->close();
+                    return;
+                }
+                BMCWEB_LOG_CRITICAL << "INFO: Hit Dump end of file";
+                this->connection->completionStatus = true;
+                this->connection->close();
+                return;
+            }
+
+            outputBuffer.commit(bytesRead);
+            auto streamHandler = [this, bytesRead, self(shared_from_this())]() {
+                this->outputBuffer.consume(bytesRead);
+                this->doReadStream();
+            };
+            this->connection->sendMessage(outputBuffer.data(), streamHandler);
+            });
+    }
+
+    std::string entryID;
+    std::string dumpType;
+    boost::beast::flat_static_buffer<socketBufferSize> outputBuffer;
+    std::filesystem::path unixSocketPath;
+    boost::asio::local::stream_protocol::socket unixSocket;
+    uint64_t dumpSize{0};
+    boost::asio::steady_timer waitTimer;
+    crow::streaming_response::Connection* connection = nullptr;
+    uint16_t connectRetryCount{0};
+};
+
+static boost::container::flat_map<crow::streaming_response::Connection*,
+                                  std::shared_ptr<Handler>>
+    bmcHandlers;
+static boost::container::flat_map<crow::streaming_response::Connection*,
+                                  std::shared_ptr<Handler>>
+    systemHandlers;
+
+inline void resetHandlers()
+{
+    if (!systemHandlers.empty())
+    {
+        auto handler = systemHandlers.begin();
+        if (handler == systemHandlers.end())
+        {
+            BMCWEB_LOG_DEBUG << "No handler to cleanup";
+            return;
+        }
+        if ((handler->second->dumpType == "system") ||
+            (handler->second->dumpType == "resource"))
+        {
+            handler->first->close();
+            BMCWEB_LOG_CRITICAL << "INFO: " << handler->second->dumpType
```

Comment:
This is really a CRITICAL log? CRITICAL is supposed to be bmcweb crashed and such

Reply:
yes correct, but we need to this info log in journal traces.
so we have decided to use CRITICAL + INFO: for debug traces